### PR TITLE
Create basic NixOS setup to build NPM dependencies locally

### PR DESCRIPTION
**Pull request by Ma27**
### Overview

Just built a basic setup to build `node2nix` dependencies in an isolated shell environment.

- [x] basic setup
- [x] fix webpack build
- [x] fix eslint

### Related issues

related to #353

### CODE REVIEW

File: Vagrantfile [hunk 1]
```diff
@@ -17,6 +17,20 @@ Vagrant.configure(2) do |config|
...
   config.vm.synced_folder '.', '/var/www/sententiaregum', :nfs => settings.fetch('nfs'), group: "www-data", mount_options: ["dmode=775,fmode=664"]
 
+  if File.symlink?('node_modules')
+    # symlink node2nix stuff into the VM
+    config.vm.synced_folder File.readlink('node_modules'), '/var/www/sententiaregum/node_modules', :nfs => settings.fetch('nfs')
+
+    # set a basic ENV parameter to prevent a second NPM install
+    config.vm.provision :shell do |s|
+      s.inline = %{
+        source ~/.profile
```
Comment (Ma27) [inline Vagrantfile]: should be `~/.bashrc`

**Comment by Ma27** [comment]
@benbieler I request a review from you (when I'm done and the `[WIP]` label is removed) to ensure that it still works on your machine :-)

**Comment by benbieler** [comment]
@Ma27 Ok fine, I'll wait until you have finished 👍

**Comment by Ma27** [comment]
kept webpack at 1.x for now (upgrading is out of scope IMHO). Need to fix the `eslint` scripts, then this is mergable :-)

**Comment by Ma27** [comment]
@benbieler I'm done here. Would be cool if you could review the dependency changes in the `package.json` and the modified code of `Vagrant` and approve this if everything works fine for you :-)

**Comment by Ma27** [comment]
If you test this it would be cool if you could ensure the following aspects:

- VM is still bootable and works (HTTP server, etc)
- local NPM builds and linting works (`npm run lint`, `npm run dev`)

to check everything I'd also recommend to kill the `node_modules` entirely as NPM produces non-deterministic results and the installations might cause conflicts with the installed stuff from our `registration` feature branch.

**Review by benbieler — APPROVED** [review]

**Comment by benbieler** [comment]
@Ma27 screen shot 2017-03-08 at 21 04 04  could this error be related to this PR?

**Comment by Ma27** [comment]
hmm, that's odd. Caused by `npm run dev`?

**Comment by Ma27** [comment]
talked to @benbieler, was just a minor `npm install` issue, merging now...

**Pull request merged**